executor/batch_point_get.go

```diff
@@ -273,6 +274,9 @@ func (e *BatchPointGetExec) initialize(ctx context.Context) error {
 		for _, idxKey := range indexKeys {
 			// lock the non-exist index key, using len(val) in case BatchGet result contains some zero len entries
 			if val := handleVals[string(idxKey)]; len(val) == 0 {
+				if !hasKeys {
+					hasKeys = true
+				}
```

Comment:
The flag should be set under `if len(lockKeys) > 0` before or after L283 , currently if all index correspond handle keys do exist the flag will not be set.